### DISCUSSION TITLE
[PSD16] Esercizio Slide 19

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#220] [Esercizi (A-L)]
![image](https://user-images.githubusercontent.com/69389461/215794310-d8f5724b-cb76-4d8a-bcb7-191c725ffddb.png)

**Comment by Elia-Belli** [comment]
### Frequenza Massima
La frequenza è l'inverso del periodo (del clock), la disciplina temporale ci impone alcuni vincoli sul periodo minimo del clock $T_c$ affinchè il circuito operi in modo affidabile, ed applicando i vari vincoli si arriva alla disequazione: 
$$T_c \geq t_{pcq} + t_{pd} + t_{su}$$
Ogni volta che si calcola un delay si valuta sempre il caso peggiore, ovvero il **critical path**, quindi in questo caso $t_{pd} = t_{gate\ pd}$, di conseguenza: $T_c \geq 120ps \rightarrow F_{max} = 1/120\ ps^{-1} = 8.3\ GHz$
### Latenza e Portata
La latenza $L$ è il tempo impiegato ad elaborare un token (un input o insieme di input), ovvero il tempo che passa dall'entrata del token nel circuito alla sua uscita, in questo caso la latenza è $L = 120\ ps$ (dal terzo token in poi), di conseguenza la portata è di $8.3 \times 10^9\ token/s$.

Rispetto agli esercizi delle Slide 17 (#218 ) e 18 (#219 ) abbiamo fatto uso del parallelismo temporale per ridurre la latenza ed aumentare la portata.
